src/core/unit_tests/reaction_ensemble_classes_test.cpp

```diff
@@ -0,0 +1,499 @@
+        auto const energy = static_cast<double>(i + 1);
+        auto const f_expr = calculate_factorial_expression(reaction, p_numbers);
+        auto const prefactor = 2e6 * f_expr;
+        auto const boltzmann = std::exp(energy / 20.);
+        double acceptance;
+        r_algo.do_energy_reweighting = false;
+        acceptance = r_algo.calculate_acceptance_probability(
+            reaction, energy, 0., p_numbers, -1, -1, false);
+        BOOST_CHECK_CLOSE(acceptance, prefactor * boltzmann, 5 * tol);
+        acceptance = r_algo.calculate_acceptance_probability(
+            reaction, energy, 0., p_numbers, -1, -1, true);
+        BOOST_CHECK_CLOSE(acceptance, boltzmann, 5 * tol);
+        r_algo.do_energy_reweighting = true;
+        acceptance = r_algo.calculate_acceptance_probability(
+            reaction, energy, 0., p_numbers, -1, -1, false);
+        BOOST_CHECK_CLOSE(acceptance, prefactor, 5 * tol);
+        acceptance = r_algo.calculate_acceptance_probability(
+            reaction, energy, 0., p_numbers, -1, -1, true);
+        BOOST_CHECK_CLOSE(acceptance, 1., 5 * tol);
+        acceptance = r_algo.calculate_acceptance_probability(
+            reaction, energy, 0., p_numbers, -1, 0, true);
+        BOOST_CHECK_CLOSE(acceptance, 10., 5 * tol);
+        acceptance = r_algo.calculate_acceptance_probability(
+            reaction, energy, 0., p_numbers, 0, -1, true);
+        BOOST_CHECK_CLOSE(acceptance, -10., 5 * tol);
+      }
+    }
+  }
+
+  // check collective variables
+  {
+    using namespace boost::adaptors;
+    // setup input
+    auto const delta_CV = 0.5;
+    auto const filename_input = std::string("wl_input.dat");
+    auto const filename_output = std::string("wl_output.dat");
+    std::ofstream wl_file;
+    wl_file.open(filename_input);
+    wl_file << "# header 1.0  0.5\n";
+    wl_file << "0  1.0  4.0\n";
+    wl_file << "1\t2.0\t5.0\n";
+    wl_file.close();
+    r_algo.do_energy_reweighting = false;
+    // add collective variable
+    r_algo.add_new_CV_potential_energy(filename_input, delta_CV);
+    BOOST_REQUIRE_EQUAL(r_algo.collective_variables.size(), 1ul);
+    BOOST_REQUIRE_EQUAL(r_algo.min_boundaries_energies.size(), 2ul);
+    BOOST_REQUIRE_EQUAL(r_algo.max_boundaries_energies.size(), 2ul);
+    BOOST_CHECK_EQUAL(r_algo.do_energy_reweighting, true);
+    BOOST_CHECK_EQUAL(r_algo.collective_variables[0]->delta_CV, delta_CV);
+    BOOST_CHECK_EQUAL(r_algo.collective_variables[0]->CV_minimum, 1.);
+    BOOST_CHECK_EQUAL(r_algo.collective_variables[0]->CV_maximum, 5.);
+    BOOST_CHECK_CLOSE(r_algo.min_boundaries_energies[0], 1., tol);
+    BOOST_CHECK_CLOSE(r_algo.min_boundaries_energies[1], 2., tol);
+    BOOST_CHECK_CLOSE(r_algo.max_boundaries_energies[0], 4., tol);
+    BOOST_CHECK_CLOSE(r_algo.max_boundaries_energies[1], 5., tol);
+    // check Wang-Landau histogram
+    {
+      r_algo.write_wang_landau_results_to_file(filename_output);
+      std::ifstream outfile;
```

Comment:
please avoid writing to disk in the tests

Reply:
This check and the corresponding RE code was changed to use a file-like object (`stringstream`) instead of an output file. The collective variable however still uses an input file that is created at the start of the test; the collective variable and Wang-Landau classes are so intertwined, that adapting them to take a file-like object would require a lot of refactoring.

Reply:
After more refactoring, the unit test no longer depends on files.